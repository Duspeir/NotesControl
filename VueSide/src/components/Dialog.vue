<script>
import { mapState } from 'pinia';
import useNotesListStore from '@/stores/NoteList';
export default {
    props: {
        note: {
            id: Number,
            title: String,
            content: String
        }
    },
    data() {
        return {
            visible: false,
            title: '',
            content: '' 
        }
    },
    computed: {
    ...mapState(useNotesListStore, ['addNote']),
    ...mapState(useNotesListStore, ['updateNote']),
    ...mapState(useNotesListStore, ['notes']),
    },
    methods:{
        show(){
            this.visible = true
        },
        close(){
            this.visible = false
        },
        changeNote(){
            this.updateNote(this.note.id, title, content)
        }
    },
    expose: ['show', 'close'],
    created(){
        if(this.note != undefined){
            this.title = this.note.title,
            this.content = this.note.content
        }
    }
}
</script>

<template>
<div v-show="visible" class="overlay">
    <div class="dialog">
        <slot />
        <textarea placeholder="Загаловок заметки..." v-model="title" id="title"/>
        <textarea rows="15" resize="none" placeholder="Текст заметки..." v-model="content" id="content"/>
        <div class="add-btn">
            <button @click="close()">Отмена</button>
            <button v-if="this.note != undefined" @click="this.updateNote(this.note.id, title, content);close()">Изменить</button>
            <button v-else @click="addNote(this.title, this.content);close()">Добавить</button>
        </div>
    </div>
</div>
</template>

<style>
.overlay{
    z-index: 100;
    width: 100%;
    height: 100%;
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
}
.dialog {
    padding: 1em;
    position: relative;
    width: 90%;
    height: 80%;
    background-color: aliceblue;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 1em;
}
.dialog > textarea {
    font-size: large;
    padding: 0.5em;
    min-height: 3em;
    width:80%;
    resize: none;
}
.add-btn {
    width: 80%;
    display: flex;
    justify-content: space-between;
}
.add-btn > button {
    font-size: large;
    border: 1px black solid;
    border-radius: 0.3em;
    padding: 5px;
    cursor: pointer;
}
.add-btn > button:hover{
    background-color: bisque;
    transform: scale(1.05);
    transition: 0.3s;
}
</style>