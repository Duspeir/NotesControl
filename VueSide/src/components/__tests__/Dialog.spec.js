import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Dialog from '../Dialog.vue'

describe('Dialog', () => {
  it('renders properly', () => {
    let wrapper = mount(Dialog, { props: {note: { id: 1, tittle:'First', content:'Message' } }})
    expect(wrapper.find('#title').exists()).toBe(true)
    expect(wrapper.find('#content').exists()).toBe(true)
    let text = wrapper.find('#content')
    expect(wrapper.vm.content === text.value).toBe(true)
    let title = wrapper.find('#title')
    expect(wrapper.vm.title === title.value).toBe(true)
    // text.setValue('Second')
    // expect(wrapper.vm.content).toBe('Second')
  })
})