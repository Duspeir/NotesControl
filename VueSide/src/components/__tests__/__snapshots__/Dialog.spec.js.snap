// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog > renders without props 1`] = `
"<div class="overlay" style="display: none;">
  <div class="dialog"><textarea placeholder="Загаловок заметки..."></textarea><textarea rows="15" resize="none" placeholder="Текст заметки..."></textarea>
    <div class="add-btn"><button>Отмена</button><button>Добавить</button></div>
  </div>
</div>"
`;
